ha/setup/cluster_layout.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+Module provides an interface for component files which describe HA cluster.
+"""
+
+import json
+from typing import Dict, List
+from abc import ABC
+from ha.core.error import HAError, HA_CLUSTER_CONFIG_ERROR
+
+
+class ClusterConfigError(HAError):
```

Comment:
I think lets move this class to error.py file, import and use here.

Reply:
That was the initial idea :)
However I had 2 reservations about it:
1. This validation module is supposed to be provided in separate `.rpm` package. So, it looks logical to ship exceptions together.
2. After reviewing `error.py` I got impression that it became a bucket of boilerplate without real need for that.
When we see `from ha.core.error import ExceptionName` - this doesn't tell a thing about who can issue such exceptions. But when approach `from ModuleName import ExceptionName` is used it becomes clear who is responsible for that exception and in which parts of current module we can expect it. 
So, I took into account 2 these things and decided keep this exception defined inside module. 
As for `error.py` role, I think make sense to keep there only exceptions that are really essential for entire HA, unless there is some requirement I'm not aware of :)

Reply:
>This validation module is supposed to be provided in separate .rpm package. So, it looks logical to ship exceptions together.
I think we are not using separate rpm for testing only but it is tested from `ha_setup test`

Can we override it from SetupError instead of HAError
ClusterConfigError It may be use by config manager for confstore as it is cluster config ot ha setup

```
After reviewing error.py I got impression that it became a bucket of boilerplate without real need for that.
When we see from ha.core.error import ExceptionName - this doesn't tell a thing about who can issue such exceptions.
```
I aggrege it has limitation and confusion for use
can we pick this topic in technical discussion and add coding guide line for this

@indrajitzagade  @azheregelya

Reply:
> Can we override it from SetupError instead of HAError

Agree. Make sense.

> ClusterConfigError It may be use by config manager for confstore as it is cluster config and ha setup

I can rename it to ValidationConfigError

Reply:
Renamed exceptions and started using SetupError.